app/controllers/questions_controller.rb

```diff
@@ -15,40 +15,57 @@ def index
     render json: questions
   end
 
+  def user_questions
+    questions = @current_user.questions
+    if questions.empty?
+      render json: [], status: :ok
+    else
+      render json: questions, status: :ok
+    end
+  end
+
   def show
-    question = Question.find(params[:id])
-    render json: question
+    answer = @question.answers
+    answer_array = Array(answer)
+    render json: { question: @question, answer: answer_array }, status: :ok
   end
 
   def update
-    if @question.user_id != @current_user.id
-      return render json: { message: 'Not authorized to update the question' },
-                    status: :unauthorized
-    end
-
     unless @question.update(question_param)
-      return render json: { message: 'Question updation failed' },
+      return render json: { message: @question.errors.full_messages },
                     status: :unprocessable_entity
     end
-
     render json: @question
   end
 
   def destroy
-    if @question.user_id != @current_user.id
-      return render json: { message: 'Not authorized to delete the question' },
-                    status: :unauthorized
+    unless @question.destroy
+      return render json: { message: @question.errors.full_messages },
+                    status: :unprocessable_entity
     end
 
-    @question.destroy
     render status: :no_content
   end
 
   private
 
+  def validate_owner
+    return unless @question.user_id != @current_user.id
+
+    message = case action_name
+              when 'update'
+                'Not authorized to update the question'
+              when 'destroy'
+                'Not authorized to delete the question'
+              end
+    render json: { message: }, status: :unauthorized
+  end
+
   def set_question
-    @question = Question.find(params[:id])
-    return render json: { message: 'Question not found' }, status: :not_found if @question.nil?
+    @question = Question.find_by(id: params[:id])
+    return unless @question.nil?
+
+    render json: { message: 'Question not found' }, status: :not_found
```

Comment:
redundant return statement here.

You can do:

render json: { message: 'Question not found' }, status: :not_found if @question.nil?